### PR TITLE
feat: attempt to enable cypress component testing

### PR DESCRIPTION
**Pull request by vdeantoni**
This is an attempt to enable component testing from cypress.

This work when running: 

``` 
npx cypress open-ct --config-file "./cypress.json"
```

from the test/components folder.

However, `bazel run test` fails with:

```
Error: SocketCt#startListening must first be called before accessing 'this.io'
    at SocketCt.ensureProp (/private/var/tmp/_bazel_vdeantoni/8f9ba4272f4c6c4105ef089cfc27042a/external/cypress_darwin/Cypress.app/Contents/Resources/app/packages/server/lib/util/class-helpers.js:14:15)
    at SocketCt.get io [as io] (/private/var/tmp/_bazel_vdeantoni/8f9ba4272f4c6c4105ef089cfc27042a/external/cypress_darwin/Cypress.app/Contents/Resources/app/packages/server/lib/socket-base.js:82:21)
    at SocketCt.toRunner (/private/var/tmp/_bazel_vdeantoni/8f9ba4272f4c6c4105ef089cfc27042a/external/cypress_darwin/Cypress.app/Contents/Resources/app/packages/server/lib/socket-base.js:88:21)
    at EventEmitter.<anonymous> (/private/var/tmp/_bazel_vdeantoni/8f9ba4272f4c6c4105ef089cfc27042a/external/cypress_darwin/Cypress.app/Contents/Resources/app/packages/server-ct/src/socket-ct.js:20:22)
    at EventEmitter.emit (events.js:315:20)
    at EventEmitter.<anonymous> (/private/var/tmp/_bazel_vdeantoni/8f9ba4272f4c6c4105ef089cfc27042a/external/cypress_darwin/Cypress.app/Contents/Resources/app/packages/server/lib/plugins/dev-server.js:19:17)
    at EventEmitter.emit (events.js:315:20)
    at ChildProcess.<anonymous> (/private/var/tmp/_bazel_vdeantoni/8f9ba4272f4c6c4105ef089cfc27042a/external/cypress_darwin/Cypress.app/Contents/Resources/app/packages/server/lib/plugins/util.js:19:22)
    at ChildProcess.emit (events.js:315:20)
    at emit (internal/child_process.js:903:12)
    at processTicksAndRejections (internal/process/task_queues.js:81:21)
```

### CODE REVIEW

File: packages/cypress/internal/run-cypress.js [hunk 1]
```diff
@@ -24,9 +24,9 @@ async function main() {
     // Detect that we are running as a test, by using well-known environment
     // variables. See go/test-encyclopedia
     if (!process.env.BUILD_WORKSPACE_DIRECTORY) {
-      await invokeCypressWithCommand('run');
+      await invokeCypressWithCommand('run-ct');
     } else {
-      await invokeCypressWithCommand('open');
+      await invokeCypressWithCommand('open-ct');
```
Comment (vdeantoni) [inline packages/cypress/internal/run-cypress.js:29]: Temporary changes, we would need to expose these different commands somehow, perhaps with a rule `cypress_ct_test`?

File: packages/cypress/test/components/cypress.json [hunk 2]
```diff
@@ -0,0 +1,8 @@
+{
+  "component": {
+    "componentFolder": ".",
```
Comment (vdeantoni) [inline packages/cypress/test/components/cypress.json:3]: I think this needs to be bazelfied... Not sure if this is the root cause.

File: packages/cypress/internal/run-cypress.js (hunk 1 above)
Comment (mrmeku) [reply packages/cypress/internal/run-cypress.js:29]: Yeah, I think your intuition to produce a new rule is a good one. That's probably the missing piece of the puzzle

File: packages/cypress/test/components/cypress.json (hunk 2 above)
Comment (mrmeku) [reply packages/cypress/test/components/cypress.json:3]: Yeah, we currently use a generated cypress plugin to add stuff to the config for test files. We will just need to do something similar for the paths of components / component tests

**Comment by alexeagle** [comment]
Ping, are you still working on this? what's the blocker?
thanks!

**Comment by vdeantoni** [comment]
> Ping, are you still working on this? what's the blocker?
> thanks!

I'm blocked. My issue is in the PR description. I'm not really sure what's wrong. My guess is the path to the test files, but I'm not sure.

**Pull request closed without merging**